### PR TITLE
Negina/TeachingDialog

### PR DESCRIPTION
**Pull request by narhami**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Added a teaching dialog that can mostly be used as the first run experience dialog

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
@dzearing does this belong in fabric internal?

**Comment by micahgodbolt** [comment]
This is a large component. We're discussing the best way to pull this in with some type of alpha/beta tag, denoting that it isn't stable yet and prone to change.

Maybe it's something we should even target for 5.0?

**Comment by dzearing** [comment]
can you include a screenshot?

File: packages/office-ui-fabric-react/src/components/TeachingDialog/TeachingDialogPage.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export class TeachingDialogPage extends React.Component<any, any> {
+  public render() {
+    return (
+      <ComponentPage
+        title='TeachingDialog'
+        componentName='TeachingDialogExample'
+        exampleCards={
+          <LayerHost>
+            <ExampleCard title='TeachingDialog' code={ TeachingDialogBasicExampleCode }>
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/TeachingDialog/TeachingDialogPage.tsx]: There is no description, dos/donts, etc. What differentiates this component from a normal dialog?

Comment (narhami) [inline packages/office-ui-fabric-react/src/components/TeachingDialog/TeachingDialogPage.tsx]: Here is a screen shot. This dialog is been used a lot with exact same colors and design as an FRE dialog. 
image

File: packages/office-ui-fabric-react/src/components/TeachingDialog/TeachingDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    for (let i: number = 0; i < this.props.viewProps.length; i++) {
+      buttons.push(
+        <IconButton
+          className={ css(styles.btn) }
+          iconProps={ { iconName: i === this.state.pageIndex ? 'circleFill' : 'CircleRing' } }
+          key={ i } />);
+    }
+
+    return buttons;
+  };
```
Comment (battletoilet) [inline packages/office-ui-fabric-react/src/components/TeachingDialog/TeachingDialog.tsx]: Unnecessary semicolon

File: packages/office-ui-fabric-react/src/components/TeachingDialog/examples/TeachingDialog.Basic.Example.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        rightButtonText: 'Done',
+        onRightButton: this._onDismiss,
+        title: 'Lorem ipsum 2',
+        textContent: 'Lorem ipsum 2 dolor sit amet, consectetur adipisicing elit. Facere, nulla, ipsum? Molestiae quis aliquam magni harum non?',
+        image: [baseProductionCdnUrl, 'teachingDialog.gif'].join('/'),
+      } as ITeachingDialogViewProps];
+
+    return (
+      <div className='ms-TeachingDialogExample'>
+        <span className='ms-TeachingDialogBasicExample-buttonArea' ref={ (menuButton) => this._menuButtonElement = menuButton }>
```
Comment (battletoilet) [inline packages/office-ui-fabric-react/src/components/TeachingDialog/examples/TeachingDialog.Basic.Example.tsx]: HTMLSpanElement | Null is not assignable to type HTMLElemement. Please change the type of this._menuButtonElement to HTMLSpanElement.

**Comment by battletoilet** [comment]
Also the dialog looks like this when i run the code locally which is propably because the image is not working. 
image

**Comment by battletoilet** [comment]
There is a bug when you run through to the end of the dialog and it closes, after you open it up again this is what appears.

**Comment by dzearing** [comment]
@narhami I think this should go into the experiments package first so that we can try it and iterate. Would you mind moving it there? I don't want this to just sit here; i think people are reluctant to include this because it's unclear if it's an atomic component which belongs in fabric-react core components, or in another package. We should try it out, iterate on it, find if it's useful and then determine if we should move.

**Comment by dzearing** [comment]
@narhami I'm going to close because it hasn't gotten traction in over 2 weeks. Please resubmit it under experiments and it can get in a lot faster! :)

**Pull request closed without merging**